src/main/java/org/opensearch/neuralsearch/highlight/SemanticHighlighterEngine.java

```diff
@@ -24,24 +28,21 @@
  * Engine class for semantic highlighting operations
  */
 @Log4j2
+@Builder
+@Value
+@ToString
+@EqualsAndHashCode
 public class SemanticHighlighterEngine {
     private static final String MODEL_ID_FIELD = "model_id";
-    private static final String DEFAULT_PRE_TAG = "<em>";
-    private static final String DEFAULT_POST_TAG = "</em>";
     private static final String MODEL_INFERENCE_RESULT_KEY = "highlights";
     private static final String MODEL_INFERENCE_RESULT_START_KEY = "start";
     private static final String MODEL_INFERENCE_RESULT_END_KEY = "end";
 
-    private final MLCommonsClientAccessor mlCommonsClient;
-    private final QueryTextExtractorRegistry queryTextExtractorRegistry;
+    @NonNull
+    MLCommonsClientAccessor mlCommonsClient;
```

Comment:
Why remove private final?

Reply:
the `@Value` annotation makes fields private and final implicitly.

Reply:
I think keeping private final would be more convenient for debugging purposes.

Reply:
Also this change is only limited to this class. Either we should adapt in the whole project or drop it.

Reply:
Sure added back.